### PR TITLE
Org.nifty-site-manager.nsm

### PR DESCRIPTION
**Pull request by n-ham**
Based on irc discussions this should now hopefully work..

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build Org.nifty-site-manager.nsm

File: org.nifty-site-manager.nsm.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "app-id": "org.flatpak.nsm",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "18.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "nsm",
+    "modules": [
+        {
+            "name": "nsm",
+            "buildsystem": "simple",
```
Comment (TingPing) [inline org.nifty-site-manager.nsm.json]: I already gave you the correct snippet here:

```json
{
  "name": "nsm",
  "no-autogen": true,
  "source": ...
}
```

You don't need to run commands manually.

File: org.nifty-site-manager.nsm.desktop [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
```
Comment (TingPing) [inline org.nifty-site-manager.nsm.desktop]: This goes upstream.

File: org.nifty-site-manager.nsm.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (TingPing) [inline org.nifty-site-manager.nsm.appdata.xml]: This goes upstream.

File: org.nifty-site-manager.nsm.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            "build-commands": [
+				"make",
+				"make install"
+            ],
+            "sources": [
+                {
+					"type": "archive",
+					"url": "https://github.com/nifty-site-manager/nsm-flatpak/raw/master/nsm.tar.xz",
```
Comment (TingPing) [inline org.nifty-site-manager.nsm.json]: This is the incorrect source, you should only have a single repo: https://github.com/nifty-site-manager/nsm

File: org.nifty-site-manager.nsm.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+<!-- Copyright 2019 Nick Ham -->
+<component type="console-application">
+  <id>org.nifty-site-manager.nsm</id>
```
Comment (TingPing) [inline org.nifty-site-manager.nsm.appdata.xml]: This app-id is invalid. It will have to be renamed to `org.nifty_site_manager.nsm`

Comment (TingPing) [reply org.nifty-site-manager.nsm.appdata.xml]: Actually you don't even control this domain the correct app-id is: `cc.nift.nsm`

File: org.nifty-site-manager.nsm.desktop [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+Version=v1.0
+Type=Application
+Terminal=true
+Name=nsm
+TryExec=nsm
```
Comment (TingPing) [inline org.nifty-site-manager.nsm.desktop]:
```suggestion
```

File: org.nifty-site-manager.nsm.desktop [hunk 7]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Version=v1.0
+Type=Application
+Terminal=true
+Name=nsm
```
Comment (TingPing) [inline org.nifty-site-manager.nsm.desktop]: Surely a human readable name here is `Nifty Site Manager`

File: org.nifty-site-manager.nsm.desktop [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+TryExec=nsm
+Exec=nsm
+Icon=org.nifty-site-manager.nsm.png
```
Comment (TingPing) [inline org.nifty-site-manager.nsm.desktop]:
```suggestion
Icon=org.nifty-site-manager.nsm
```

**Comment by TingPing** [comment]
This is also a CLI application not a TUI application which traditionally have not gone on Flathub (with a few exceptions).

I personally still don't believe they belong here but I won't block it if others do.

The real problem then is that your desktop file is nonsensical knowing that and should be deleted and then at that point it will never show up in any store

File: org.nifty-site-manager.nsm.json (hunk 4 above)
Comment (n-ham) [reply org.nifty-site-manager.nsm.json]: I would have to put files for flathub/snapcraft/chocolatey/homebrew if it was all in the same repo which is a little confusing for people. Seemed nicer for me to just have a separate repo for each platform?

File: org.nifty-site-manager.nsm.json (hunk 1 above)
Comment (n-ham) [reply org.nifty-site-manager.nsm.json]: I'm not quite sure what you're suggesting here? 

Replace buildsystem with no-autogen and sources? Is that ... for sources what I put or am I meant to know what to put there? Sorry I have not released on FlatHub before..

Comment (n-ham) [reply org.nifty-site-manager.nsm.json]: Oh wait, get rid of buildsystem and build-commands?

Comment (TingPing) [reply org.nifty-site-manager.nsm.json]: I'm saying remove "buildsystem" and "build-commands", add "no-autogen".

File: org.nifty-site-manager.nsm.json (hunk 4 above)
Comment (TingPing) [reply org.nifty-site-manager.nsm.json]: Nothing is unique to Flatpak. It should be building your normal sources.

**Pull request closed without merging**